File: Command/Python/TurnOffTv.py
# -*- coding: utf-8 -*-
"""
Created on Wed May 18 21:03:45 2022

@author: Mahmoud Saeed
"""

import Command as command

class TurnOffTv(command.Command):
    tv = None
    
    def __init__(self , tv):
        self.tv = tv 
        
    def excute(self):
        self.tv.turnoff()
    
    def undo(self):
        self.tv.turnon()
        

File: Command/Java/src/command/turnOffTv.java
package command;

public class turnOffTv implements Command{
    private Tv tv;
    public turnOffTv(Tv tv){
        this.tv = tv;
    }


    @Override
    public void execute() {
        this.tv.turnOff();
    }

    @Override
    public void undo() {
        this.tv.turnOn();

    }
}
